schema.json

```diff
@@ -0,0 +1,118 @@
+  "title": "Thunderbird In-App Notifications",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "type": "string",
+        "format": "uuid",
+        "description": "Unique ID set by the server"
+      },
+      "start_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "Timestamp after which Thunderbird will show the event after startup."
+      },
+      "end_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "Timestamp after which Thunderbird will never show the event, even if it has never been shown to the user."
+      },
+      "title": {
+        "type": "string",
+        "description": "Short sentence describing the event which will be displayed in the Thunderbird UI."
+      },
+      "description": {
+        "type": "string",
+        "description": "A short paragraph that can contain HTML and will be displayed in the Thunderbird UI."
+      },
+      "URL": {
+        "type": "string",
+        "format": "uri",
+        "description": "URL to load, if any."
+      },
+      "CTA": {
+        "type": "string",
+        "description": "Link text to show for the URL."
+      },
+      "severity": {
+        "enum": [1, 2, 3, 4, 5],
+        "description": "Severity level, where 1 is the most important/urgent and 5 is the least."
+      },
+      "type": {
+        "enum": ["donation", "message", "security", "blog"],
+        "description": "Category of notification."
+      },
+      "targeting": {
+        "type": "object",
+        "properties": {
+          "percent_chance": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 100
+          },
+          "exclude": {
+            "$ref": "#/definitions/profile"
+          },
+          "include": {
+            "$ref": "#/definitions/profile"
+          }
+        }
+      }
+    },
+    "required": [
+      "id",
+      "start_at",
+      "end_at",
+      "title",
+      "description",
+      "severity",
+      "type",
+      "targeting"
+    ],
+    "dependentRequired": { "CTA": ["URL"] }
+  },
+  "definitions": {
+    "profile": {
+      "type": "object",
+      "properties": {
+        "locales": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "versions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "channels": {
+          "type": "array",
+          "items": {
+            "enum": ["esr", "release", "beta", "daily"],
+            "description": "Channel"
+          }
+        },
+        "operating_systems": {
+          "type": "array",
+          "items": {
+            "enum": [
+              "windows",
+              "macos",
+              "linux",
+              "freebsd",
+              "openbsd",
+              "netbsd",
+              "solaris"
```

Comment:
I think we could also encounter an "other" value in the wild here.

Reply:
great suggestion!
I'll add that